swagger/swaggerDoc.yaml

```diff
@@ -0,0 +1,849 @@
+  /api/web_app_config:
+    get:
+      parameters: []
+      responses:
+        200:
+          description: "Status 200"
+    put:
+      parameters: []
+      responses:
+        200:
+          description: "Status 200"
+    delete:
+      parameters: []
+      responses:
+        200:
+          description: "Status 200"
+  /app/{imageName}:
+    get:
+      description: "imageName=nope.jpg"
+      parameters: []
+      responses:
+        200:
+          description: "Status 200"
+    parameters:
+    - name: "imageName"
+      in: "path"
+      required: true
+      type: "string"
+  /{jsonFile}:
+    post:
+      description: "jsonFile=csp_reports.json"
+      produces:
+      - "application/json"
+      parameters: []
+      responses:
+        200:
+          description: "Status 200"
+          schema:
+            type: "object"
+          examples:
+            application/json: "{\n  \n}"
+        400:
+          description: "Status 400"
+          schema:
+            type: "object"
+            example: "{\n  \"problem\": \"Crashed while parsing report\"\n}"
+    parameters:
+    - name: "jsonFile"
+      in: "path"
+      required: true
+      type: "string"
+  /tos_update:
+    get:
+      description: "???\nwhat is a tos_update??"
+      parameters: []
+      responses:
+        200:
+          description: "Status 200"
+  /api/corpuses/{IntId}:
+    get:
+      summary: "Corpuse or Corpuses"
+      description: "* with IntId-Variable:  get one corpuse.\n* no IntId-Variable:\
+        \ get all corpuses"
+      produces:
+      - "application/json"
+      parameters: []
+      responses:
+        200:
+          description: "Status 200"
+          schema:
+            type: "array"
+            items:
+              $ref: "#/definitions/corpuse"
+    parameters:
+    - name: "IntId"
+      in: "path"
+      required: true
+      type: "string"
+  /api/device:
+    get:
+      parameters: []
+      responses:
+        200:
+          description: "Status 200"
+          schema:
+            $ref: "#/definitions/device"
+    post:
+      consumes: []
+      parameters:
+      - name: "body"
+        in: "body"
+        required: true
+        schema:
+          type: "object"
+          properties:
+            name:
+              type: "string"
+          example: "{\n  \"name\": \"maculate-moon-349\"\n}"
+      responses:
+        200:
+          description: "Status 200"
+          schema:
+            $ref: "#/definitions/device"
+    delete:
+      description: "IMHO - cant work - how the server should know which device has\
+        \ to be deleted?"
+      produces:
+      - "text/plain"
+      parameters: []
+      responses:
+        200:
+          description: "Status 200"
+          schema:
+            type: "object"
+            description: "Empty Response\n"
+  /api/device/{IntId}:
+    put:
+      description: "why the server needs the farmbot-id and another attribute (e.g\
```

Comment:
This endpoint can be ignored- it's there for the convenience of developers that are using REST adapters that always expect an ID. The "proper" way to use this endpoint is to not provide an ID.

Most resources on the FarmBot API are "plural" resources. Eg, you can have many of them per account.

Examples:

  * farm_events
 * images
 * logs
 * peripherals
 * sequences

There are still some singular resources which an API user will only have one of:

 * device
 * web_app_config
 * public_key

Sometimes users (or the tools they use) expect the URL to contain an ID in the form of `/api/RESOURCE_NAME/RESOURCE_ID` even though, in the case of singular resources, it does not make sense.

For convenience, we allow singular resources to have an ID even though it is redundant. I've considered removing this feature but have not had the time to go back and fix it.